### PR TITLE
feat: Show past events

### LINKED ISSUE #52: [Feature Request] Show past events
### Overview

Currently, past events are still shown under "upcoming events" which is not ideal. Once a lot of events have occurred, the past events would still show up, making it harder to find the actual upcoming events. We should either show past events in a separate section or remove past events entirely from view.

### Details

The date and time of the event can be parsed to determine if the event has occurred or not.

### Why would this feature be useful?

So people can see the events that actually haven't occurred yet first, then any past events down below.

### Additional Notes

_No response_

### PR DESCRIPTION
**Pull request by phoenixpereira**
### Description
Shows past events on the events page.

### Changes Made
- Sorts events into upcoming or past based on date and time.
- Conditionally renders past events title and events if there are past events.
- Adds a greyscale effect to past events to further differentiate them.

### Related Issues
Fixes #52.

### Additional Notes
N/A

### CODE REVIEW

File: src/app/events/Events.tsx [hunk 1]
```diff
@@ -26,7 +34,7 @@ function EventCard({ event, index }: { event: Event; index: number }) {
                     alt={`${event.title}`}
                     width={450}
                     height={450}
-                    className="w-full shrink-0 rounded-lg border-[3px] border-black bg-white object-contain md:w-[450px]"
+                    className={`w-full shrink-0 rounded-lg border-[3px] border-black bg-white object-contain md:w-[450px] ${isPastEvent ? 'grayscale' : ''}`}
```
Comment (jsun969) [inline src/app/events/Events.tsx:37]: use clsx

File: src/app/events/Events.tsx [hunk 2]
```diff
@@ -44,7 +52,7 @@ function EventCard({ event, index }: { event: Event; index: number }) {
                             </div>
                         </div>
                         <div
-                            className={`h-fit rounded-md border-[3px] border-black px-4 py-2 ${['bg-orange', 'bg-yellow', 'bg-purple'][index % 3]}`}
+                            className={`h-fit rounded-md border-[3px] border-black px-4 py-2 ${['bg-orange', 'bg-yellow', 'bg-purple'][index % 3]} ${isPastEvent ? 'grayscale' : ''}`}
```
Comment (jsun969) [inline src/app/events/Events.tsx:55]: use clsx

File: src/app/events/Events.tsx [hunk 3]
```diff
@@ -58,12 +66,39 @@ function EventCard({ event, index }: { event: Event; index: number }) {
 }
 
 export default function Events({ className }: { className?: string }) {
+    const currentDate = new Date(); // Current timestamp
+
+    // Convert event date and check if upcoming event
+    const upcomingEvents = EVENTS.filter((event) => {
+        const eventDate = new Date(
```
Comment (jsun969) [inline src/app/events/Events.tsx:73]: add a util function like `getEventDate`

File: src/app/events/Events.tsx [hunk 4]
```diff
@@ -3,21 +3,29 @@ import { EVENTS, type Event } from '@/data/events';
 import Image from 'next/image';
 import { FiClock, FiMapPin } from 'react-icons/fi';
 
-function Title() {
+function Title({ title }: { title: string }) {
```
Comment (jsun969) [inline src/app/events/Events.tsx:6]:
```suggestion
function Title({ children }: { children: string }) {
```

**Review by jsun969 — CHANGES_REQUESTED** [review]

File: src/utils/get-events-date.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const currentDate = new Date();
+    return EVENTS.filter((event) => {
+        const eventDate = new Date(
+            `${event.date.year} ${event.date.month} ${event.date.day} ${event.date.endTime}`
+        );
+        return eventDate >= currentDate;
+    });
+}
+
+// Convert event date and return past events in reversed order
```
Comment (rayokamoto) [inline src/utils/get-events-date.ts:14]: By "reversed order" do you mean the most recent event is at the top? If so, maybe make this clearer in the comment

**Review by rayokamoto — CHANGES_REQUESTED** [review]

**Review by rayokamoto — APPROVED** [review]

**Review by jsun969 — APPROVED** [review]

**Pull request merged**